packages/react-core/src/components/CodeBlock/examples/CodeBlock.md

```diff
@@ -0,0 +1,128 @@
+---
+id: Code block
+section: components
+cssPrefix: pf-c-code-block
+propComponents: ['CodeBlock', 'CodeBlockAction', 'CodeBlockCode']
+beta: true
+---
+
+import CopyIcon from '@patternfly/react-icons/dist/js/icons/copy-icon';
+import PlayIcon from '@patternfly/react-icons/dist/js/icons/play-icon';
+
+## Examples
+
+### Basic
+
+```js
+import React from 'react';
+import { CodeBlock, CodeBlockAction, CodeBlockCode, Button } from '@patternfly/react-core';
+import CopyIcon from '@patternfly/react-icons/dist/js/icons/copy-icon';
+import PlayIcon from '@patternfly/react-icons/dist/js/icons/play-icon';
+
+class BasicCodeBlock extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const actions = (
+      <React.Fragment>
+        <CodeBlockAction>
+          <Button variant="plain" aria-label="Copy icon">
+            <CopyIcon />
+          </Button>
+        </CodeBlockAction>
+        <CodeBlockAction>
+          <Button variant="plain" aria-label="Play icon">
+            <PlayIcon />
+          </Button>
+        </CodeBlockAction>
+      </React.Fragment>
+    );
+    return (
+      <CodeBlock actions={actions}>
+        <CodeBlockCode>
+          apiVersion: helm.openshift.io/v1beta1/
+          <br />
```

Comment:
Ideally we wouldn't need any markup for formatting in this block. `<CodeBlockCode>` is configured to preserve white space, so that a line break in the markup would produce a line break in the browser.

However, due to this, if we add structural elements inside of this block (like the expandable content in [this core example](https://pf4.patternfly.org/components/code-block#expandable)), we will need to remove new lines and space characters so that we don't create unnecessary space like you see in the linked core example.

So basically we need to preserve all space/line breaks within the code a user passes, but remove the leading/trailing space/line breaks around it and around any structure elements we add within `<CodeBlockCode>`.

Reply:
Sounds tricky. Is there anything I should update for the moment on my end?

Reply:
Are we able to preserve white space within the children of `<CodeBlockCode>`?

Reply:
Yes, conditionally. If you took the whole block of text and sent it in as a formatted/spaced string it would work. If you pass in just the text directly without the `<br />` tags though, it compiles down to a single line.

Something like:
```
const codeStr = `apiVersion: helm.openshift.io/v1beta1/
kind: HelmChartRepository
metadata:
name: azure-sample-repo
spec:
connectionConfig:
url: https://raw.githubusercontent.com/Azure-Samples/helm-charts/master/docs`;
```

I'm not sure how you would break up the formatted string when it comes to expandable sections, maybe you'd go back to markup.

Reply:
@kmcfaul that works great! Based off of these comments, maybe we should update our examples to pass the text in as a formatted string?

https://github.com/patternfly/patternfly/pull/3937#issuecomment-809395629
https://github.com/patternfly/patternfly/pull/3937#issuecomment-809437273

Reply:
Sure thing. I will update

Reply:
I will also update the detached expandable section example once that goes in.